### PR TITLE
Create new userspace for shurikai

### PR DESCRIPTION
**Pull request by shurikai**
Create new userspace for my personal QMK configuration.

## Description
Add new userspace, along with one keymap (niu_mini) that depends on it.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/niu_mini/keymaps/shurikai/config.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#pragma once
```
Comment (drashna) [inline keyboards/niu_mini/keymaps/shurikai/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/niu_mini/keymaps/shurikai/config.h (hunk 1 above)
Comment (shurikai) [reply keyboards/niu_mini/keymaps/shurikai/config.h:1]: Done.

Comment (shurikai) [reply keyboards/niu_mini/keymaps/shurikai/config.h:1]: Should I change headers in other files as well?

Comment (drashna) [reply keyboards/niu_mini/keymaps/shurikai/config.h:1]: Does not look like it took.  Did you push the changes

File: users/shurikai/wrappers.h [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+#if (!defined(LAYOUT) && defined(KEYMAP))
+#    define LAYOUT KEYMAP
+#endif
+
+#define LAYOUT_wrapper(...)          LAYOUT(__VA_ARGS__)
+#define LAYOUT_reviung_wrapper(...)  LAYOUT_reviung39(__VA_ARGS__)
+#define LAYOUT_microdox_wrapper(...) LAYOUT_split_3x5_3(__VA_ARGS__)
+#define LAYOUT_corne_wrapper(...)    LAYOUT_split_3x6_3(__VA_ARGS__)
+#define LAYOUT_preonic_wrapper(...)  LAYOUT_preonic_grid(__VA_ARGS__)
+#define LAYOUT_niumini_wrapper(...)  LAYOUT_ortho_4x12(__VA_ARGS__)
```
Comment (drashna) [inline users/shurikai/wrappers.h:28]: Ideally, these should o in the keymap.c files for their specific boards

File: keyboards/niu_mini/keymaps/shurikai/config.h (hunk 1 above)
Comment (drashna) [reply keyboards/niu_mini/keymaps/shurikai/config.h:1]: > Should I change headers in other files as well?

Only on c and h files. rules.mk don't need them

**Pull request closed without merging**